components/script/dom/htmlelement.rs

```diff
@@ -467,52 +469,90 @@ impl HTMLElementMethods for HTMLElement {
         DOMString::from(text)
     }
 
-    // https://html.spec.whatwg.org/multipage/#the-innertext-idl-attribute
+    /// <https://html.spec.whatwg.org/multipage/#set-the-inner-text-steps>
     fn SetInnerText(&self, input: DOMString) {
+        // Step 1: Let fragment be the rendered text fragment for value given element's node
+        // document.
+        let fragment = self.rendered_text_fragment(input);
+
+        // Step 2: Replace all with fragment within element.
+        Node::replace_all(Some(fragment.upcast()), self.upcast::<Node>());
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#dom-outertext>
+    fn GetOuterText(&self) -> Fallible<DOMString> {
```

Comment:
I think it's worth adding a method for [`get-the-text-steps`](https://html.spec.whatwg.org/multipage/dom.html#get-the-text-steps), and use it here and within `GeInnerText`.

Reply:
Makes sense, they were treated separately until they hit Layout where both ended up calling the same method, I've added a `get_inner_outer_text` method to HTMLElement which is called by both `GetInnerText` and `GetOuterText`, I didn't want call this method `get_the_text_steps` since the actual implementation is in Layout, where the method is called just that.